components/docusign/docusign.app.js

```diff
@@ -0,0 +1,129 @@
+const axios = require("axios");
+
+module.exports = {
+  type: "app",
+  app: "docusign",
+  propDefinitions: {
+    account: {
+      type: "string",
+      label: "Account",
+      description: "",
+      async options() {
+        const { accounts } = await this.getUserInfo();
+        return accounts.map((account) => {
+          return {
+            label: account.account_name,
+            value: account.account_id,
+          };
+        });
+      },
+    },
+    template: {
+      type: "string",
+      label: "Template",
+      description: "",
+      async options({ account }) {
+        const baseUri = await this.getBaseUri(account);
+        const { envelopeTemplates } = await this.listTemplates(
+          baseUri,
+          account,
+        );
+        return envelopeTemplates.map((template) => {
+          return {
+            label: template.name,
+            value: template.templateId,
+          };
+        });
+      },
+    },
+    role: {
+      type: "string",
+      label: "Recipient Role",
+      description: "",
+      async options({
+        account,
+        template,
+      }) {
+        const baseUri = await this.getBaseUri(account);
+        const { signers } = await this.listTemplateRecipients(
+          baseUri,
+          account,
+          template,
+        );
+        return signers.map((signer) => {
+          return signer.roleName;
+        });
+      },
+    },
+    emailSubject: {
+      type: "string",
+      label: "Email Subject",
+      description: "Subject line of email",
+    },
+    emailBlurb: {
+      type: "string",
+      label: "Email Blurb",
+      description: "Email message to recipient. Overrides template setting.",
+      optional: true,
+    },
+    recipientEmail: {
+      type: "string",
+      label: "Recipient Email",
+      description: "Email address of signature request recipient",
+    },
+    recipientName: {
+      type: "string",
+      label: "Recipient Name",
+      description: "The full name of the recipient",
+    },
+  },
+  methods: {
+    _getHeaders() {
+      return {
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        "Content-Type": "application/json",
+      };
+    },
+    async _makeRequest(method, url, data = null, params = null) {
+      const config = {
+        method,
+        url,
+        headers: this._getHeaders(),
+        data,
+        params,
+      };
+      return (await axios(config)).data;
+    },
+    async getUserInfo() {
+      return await this._makeRequest(
+        "GET",
+        "https://account-d.docusign.com/oauth/userinfo",
```

Comment:
Is this URI distinct from the base URI retrieved for the user's account below?

Reply:
Yes. For developer accounts, the Base URI is https://demo.docusign.net.